### PR TITLE
Backport Label3D node implementation and Sprite*3D material render priority.

### PR DESCRIPTION
**Pull request by bruvzg**
3.x backport of #60386, #60595 and #60650. Using a copy of `Label` text processing logic instead of `TextServer`.

Screenshot 2022-05-22 at 13 19 57

### CODE REVIEW

File: editor/spatial_editor_gizmos.h [hunk 1]
```diff
@@ -126,6 +126,19 @@ class Sprite3DSpatialGizmoPlugin : public EditorSpatialGizmoPlugin {
 	Sprite3DSpatialGizmoPlugin();
 };
 
+class Label3DSpatialGizmoPlugin : public EditorSpatialGizmoPlugin {
+	GDCLASS(Label3DSpatialGizmoPlugin, EditorSpatialGizmoPlugin);
+
+public:
+	bool has_gizmo(Spatial *p_spatial);
+	String get_gizmo_name() const;
```
Comment (timothyqiu) [inline editor/spatial_editor_gizmos.h:134]: This function is `get_name()` on `3.x`.

File: scene/3d/label_3d.cpp [hunk 2]
```diff
@@ -0,0 +1,1049 @@
...
+	faces.resize(6);
+	PoolVector<Vector3>::Write facesw = faces.write();
+
+	if (word_cache_dirty) {
+		const_cast<Label3D *>(this)->regenerate_word_cache();
+	}
+
+	int font_h = (font->get_height() + line_spacing);
+	real_t space_w = font->get_char_size(' ').width;
+	float total_h = line_count * font_h;
```
Comment (timothyqiu) [inline scene/3d/label_3d.cpp:419]: Line spacing should not be added on the last line I think.

e.g. If there's only one line, changing the line spacing currently affects the text's rendered position.

Comment (bruvzg) [reply scene/3d/label_3d.cpp:419]: This makes sense. I guess `Label` have the same issue.

Comment (timothyqiu) [reply scene/3d/label_3d.cpp]: Ah, I checked `Label` and this code again, it seems fine if `total_h` is used as `total_h - line_spacing` later.

So the offset when changing line spacing might be caused by other codes related to this.

**Comment by Calinou** [comment]
I gave this PR a quick try locally. It works as expected :slightly_smiling_face:

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**